### PR TITLE
In place union and concatenation in Python binding

### PR DESCRIPTION
**Pull request by Adda0**
This PR adds in-place variants of concatenation and union to Python binding. Additionally, it adds a temporary deepcopy function for Python `Nfa` class. Proper deepcopy using Python `copy.deepcopy()` will come in a later PR.

### CODE REVIEW

File: bindings/python/libmata/nfa/nfa.pyx [hunk 1]
```diff
@@ -526,6 +533,22 @@ cdef class Nfa:
         self.thisptr.get().trim(&state_map)
         return self, {k: v for k, v in state_map}
 
+    def concatenate(self, Nfa other) -> Nfa:
+        """Concatenate self with other in-place of self.
```
Comment (jurajsic) [inline bindings/python/libmata/nfa/nfa.pyx]: I don't understand this sentence

File: bindings/python/libmata/nfa/nfa.pyx [hunk 2]
```diff
@@ -526,6 +533,22 @@ cdef class Nfa:
...
+
+        :return: Self
+        """
+        self.thisptr.get().concatenate(dereference(other.thisptr.get()))
+        return self
+
+    def union(self, Nfa other) -> Nfa:
+        """Compute union of self with other in-place of self.
```
Comment (jurajsic) [inline bindings/python/libmata/nfa/nfa.pyx]: Also this sentence

**Review by jurajsic — APPROVED** [review]
Looks good, just some weird sentences in comments

File: bindings/python/libmata/nfa/nfa.pyx (hunk 1 above)
Comment (Adda0) [reply bindings/python/libmata/nfa/nfa.pyx]: Reformulated.

File: bindings/python/libmata/nfa/nfa.pyx (hunk 2 above)
Comment (Adda0) [reply bindings/python/libmata/nfa/nfa.pyx]: Improved.

**Pull request merged**